src/tools/ScenarioMeasurement/Startup/TraceSourceManager.cs

```diff
@@ -55,7 +57,7 @@ public interface IKernelParser
 
     public sealed class LinuxKernelParser : IKernelParser
     {
-        LinuxKernelEventParser parser;
+        readonly LinuxKernelEventParser parser;
```

Comment:
Are we adding ```readonly``` here to make the variable immutable? If so what's the rule of predicting it will never be re-assigned outside the constructor?

Reply:
Correct, it makes it immutable because the only place where we assign a value to this object is in the constructor. This IDE suggestion won't show up if the object is assigned other values in other places that are not the constructor.